crates/nix_rs/src/flake/outputs.rs

```diff
@@ -6,35 +6,71 @@ use std::{
     fmt::Display,
 };
 
-/// Absolute path to the `nix` binary compiled with flake schemas support
-///
-/// We expect this environment to be set in Nix build and shell.
-pub const NIX_FLAKE_SCHEMAS_BIN: &str = env!("NIX_FLAKE_SCHEMAS_BIN");
+use crate::system_list::SystemsListFlakeRef;
 
 /// Flake URL of the default flake schemas
 ///
 /// We expect this environment to be set in Nix build and shell.
 pub const DEFAULT_FLAKE_SCHEMAS: &str = env!("DEFAULT_FLAKE_SCHEMAS");
 
+/// Flake URL of the flake that defines functions for inspecting flake outputs
+///
+/// We expect this environment to be set in Nix build and shell.
+pub const INSPECT_FLAKE: &str = env!("INSPECT_FLAKE");
+
 /// Represents the "outputs" of a flake
 ///
-/// This structure is currently produced by `nix flake show`, thus to parse it we must toggle serde untagged.
+/// TODO: Rename this to `FlakeSchema` while generalizing the existing `schema.rs` module.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub enum FlakeOutputs {
+pub struct FlakeOutputs {
+    pub inventory: BTreeMap<String, InventoryItem>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum InventoryItem {
     Leaf(Leaf),
-    Attrset(BTreeMap<String, FlakeOutputs>),
+    Attrset(BTreeMap<String, InventoryItem>),
 }
 
 impl FlakeOutputs {
-    /// Run `nix flake show` on the given flake url
+    /// Determine flake outputs using [INSPECT_FLAKE] and [DEFAULT_FLAKE_SCHEMAS]
     pub async fn from_nix(
         nix_cmd: &crate::command::NixCmd,
         flake_url: &super::url::FlakeUrl,
+        system: &super::System,
     ) -> Result<Self, crate::command::NixCmdError> {
-        let v = FlakeOutputsUntagged::from_nix(nix_cmd, flake_url).await?;
-        Ok(v.into_flake_outputs())
+        let v = nix_cmd
+            .run_with_args_expecting_json(&[
+                "eval",
+                "--json",
+                "--override-input",
+                "flake-schemas",
+                env!("DEFAULT_FLAKE_SCHEMAS"),
+                "--override-input",
+                "flake",
+                flake_url,
+                "--override-input",
+                "systems",
+                // TODO: don't use unwrap
+                &SystemsListFlakeRef::from_known_system(system).unwrap().0,
+                "--no-write-lock-file",
+                // Why `exculdingOutputPaths`?
+                //   This function is much faster than `includingOutputPaths` and also solves <https://github.com/juspay/omnix/discussions/231>
```

Comment:
Note to self: I should improve `nix_rs` command invocation API. And make use of this struct,

https://github.com/juspay/omnix/blob/b5f3b54e4a290372889dcceabd8ed48573f05d5f/crates/nix_rs/src/flake/command.rs#L70-L90

Reply:
#280